errbot/backends/slack.py

```diff
@@ -160,7 +160,8 @@ def __str__(self):
         return self.__unicode__()
 
     def __eq__(self, other):
-        return other.userid == self.userid
+        return other.userid == self.userid and \
+               other.channelid == self.channelid
```

Comment:
What you wanted to do maybe is to differentiate a RoomOccupant from a Person then testing the type of other is better.